src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,288 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/ScalarWave/BoundaryCorrections/BoundaryCorrection.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ScalarWave::BoundaryCorrections {
+/*!
+ * \brief Computes the scalar wave upwind multipenalty boundary
+ * correction.
+ *
+ * This implements the upwind multipenalty boundary correction term
+ * \f$D_\alpha\f$. The general form is given by:
+ *
+ * \f{align*}{
+ *   \label{eq:pnpm upwind boundary term characteristics}
```

Comment:
Why the label? Are you planning to refer to this equation from elsewhere in the dox?